AK/Time.h

```diff
@@ -453,6 +458,66 @@ struct UnixDateTime : Detail::NaiveTime {
     }
 };
 
+// Monotonic time represents time returned from the CLOCK_MONOTONIC clock, which has an arbitrary fixed reference point.
+struct MonotonicTime : private Detail::NaiveTime {
+    // Monotonic time does not have a defined reference point.
+    // A MonotonicTime at the reference point is therefore meaningless.
+    MonotonicTime() = delete;
+    constexpr MonotonicTime(MonotonicTime const&) = default;
+    constexpr MonotonicTime(MonotonicTime&&) = default;
+    constexpr MonotonicTime& operator=(MonotonicTime const&) = default;
+    constexpr MonotonicTime& operator=(MonotonicTime&&) = default;
+
+#ifndef KERNEL
+    [[nodiscard]] static MonotonicTime now();
+    [[nodiscard]] static MonotonicTime now_coarse();
+#endif
+
+    /* clang-format off */ // Clang-format likes to expand this function for some reason.
+    [[nodiscard]] i64 seconds() const { return offset.to_seconds(); }
+    /* clang-format on */
+    [[nodiscard]] i64 milliseconds() const { return offset.to_milliseconds(); }
+    [[nodiscard]] i64 nanoseconds() const { return offset.to_nanoseconds(); }
+    // Never returns a point in the future, since fractional seconds are cut off.
+    [[nodiscard]] i64 truncated_seconds() const { return offset.to_truncated_seconds(); }
+    [[nodiscard]] i64 nanoseconds_within_second() const { return to_timespec().tv_nsec; }
+
+    constexpr bool operator==(MonotonicTime const& other) const
+    {
+        return this->offset == other.offset;
+    }
```

Comment:
If we are going the `clang-format off` way, this might also want that treatment?